packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -3,19 +3,29 @@ import 'dart:html' as html;
 
 import 'package:flutter/services.dart';
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:google_sign_in_web/src/gapi.dart';
+import 'package:js/js.dart';
+
+const String _kClientIdMetaSelector = 'meta[name=google-signin-client_id]';
+const String _kClientIdAttributeName = 'content';
+const List<String> _kJsLibraries = <String>[
+  'https://apis.google.com/js/platform.js'
+];
 
 /// Implementation of the google_sign_in plugin for Web
 class GoogleSignInPlugin {
   GoogleSignInPlugin() {
-    _jsLibrariesLoading = _injectJSLibraries(<String>[
-      'https://apis.google.com/js/platform.js',
-    ])..then(_initGapi);    
+    _autoDetectedClientId = html
+        .querySelector(_kClientIdMetaSelector)
+        ?.getAttribute(_kClientIdAttributeName);
+
+    _isGapiInitialized =
+        Future.wait(_injectJSLibraries(_kJsLibraries)).then(_initGapi);
```

Comment:
once you change `_injectJSLibraries` to return `Future<void>` you can make this into just `_injectJSLibraries(_kJsLibraries).then(_initGapi)`

Reply:
once you remove the unused argument in `initGapi` you can do `.then((_) => _initGapi())` which is the more common pattern

Reply:
After some ruminating, the second part makes sense; done :P